### ISSUE TITLE
Exception thrown for empty 'tool' field on URL import.

### ISSUE THREAD

**Issue by Jaymes92** [Teifun2/nextcloud-cookbook-flutter#147]
When importing a recipe through the URL that has an empty 'tool' field, when you click to display it in the app you get an error:

Exception: NoSuchMethodError: The method 'cast' was called on null.
Receiver: null
Tried calling: cast<String>()

You can enter any arbitrary entry into the 'Tool' field and this error will go away. 

My guess is this has to do with the /lib/src/models/recipe.dart lines 105-110 (but not certain, I'm new to this and do not know how to build/test these changes).

List<String> tool = [];
    if (data["tool"] is Map) {
      data["tool"].forEach((k, v) => tool.add(v as String));
    } else {
      tool = data["tool"].cast<String>().toList();
    }

It seems like if it is empty, the else statement is throwing this error.

Thank you for this app!

**Comment by Teifun2** [comment]
Hello Thanks for the report.

I cannot reproduce the issue. Can you send me the link of the recipe you tried it with?

**Comment by Jaymes92** [comment]
Thanks for the reply! 

It seems to happen on most any that I import from a URL, I just worked around the problem by putting in "N/A" for the "Tools" entry in my cookbook Nextcloud app.

For example, both of the following recipes will throw the error if they are imported as-is: 
https://www.allrecipes.com/recipe/286244/stuffed-chicken-breasts-with-bacon-and-cream-cheese/
https://www.budgetbytes.com/vegetable-barley-soup/

I have attached the corresponding recipe.json files that is generated in my Notes directory for these two as examples.  

[recipe_json_examples.tar.gz](https://github.com/Teifun2/nextcloud-cookbook-flutter/files/9522544/recipe_json_examples.tar.gz)

I will clarify that the error for these two says "Tried calling: cast<String>()" instead of the "Tried calling: cast()" in the initial issue above.
